step-maven-plugin/src/main/java/step/plugins/maven/DeployAutomationPackageMojo.java

```diff
@@ -66,9 +70,9 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 
             getLog().info("Uploading the automation package...");
             try {
-                String uploadedId = automationPackageClient.createOrUpdateAutomationPackage(packagedTarget);
-                if (uploadedId != null) {
-                    getLog().info("Automation package successfully uploaded. Id: " + uploadedId);
+                AutomationPackageUpdateResult updateResult = automationPackageClient.createOrUpdateAutomationPackage(packagedTarget, async != null && async);
+                if (updateResult != null && updateResult.getId() != null) {
```

Comment:
Why check for null? Isn't there always a result returned if there's no exception?